site/_posts/2017-10-24-api-contributors-guide.md

```diff
@@ -0,0 +1,252 @@
+              :identifier: coffee_edit 
+            - :name: delete
+              :identifier: coffee_delete
+        :resource_actions:
+            :get:
+            - :name: read
+              :identifier: coffee_show 
+            :post:
+            - :name: edit
+              :identifier: coffee_edit 
+            - :name: delete
+              :identifier: coffee_delete
+            :delete:
+            - :name: delete
+              :identifier: coffee_delete
+```
+
+
+Below is a breakdown of what some of the above elements mean:
+* **: coffee:**
+    * Defines the collection or subcollection portion of the URI in conjunction with the name and supported verbs (in our case, `GET api/coffee`, `POST /api/coffee`, etc..)
+    * Tells the API that the controller will be `CoffeeController`
+* **:options:**
+    * Gives more details about the collection and whether it will act as a collection, subcollection
+* **:verbs:**
+    * Specifies the [verb set](https://github.com/ManageIQ/manageiq-api/blob/4811ec9fcea2424a585560a761e05d56c53535f8/config/api.yml#L18-L36) that can be used on the collection
+* **:klass:** 
+    * Specifies the base class name of the resources under this collection 
+* **:name:**
+    * Name of the action to be performed 
+* **:identifier:**
+    * Name of the corresponding [product feature](https://github.com/ManageIQ/manageiq/blob/f77c75b4659f97be95f8976ef58abe521085c963/db/fixtures/miq_product_features.yml), which is used to control permissions to resources
+* **:collection_actions:** vs **:resource_actions:**
+    * collection actions correspond to those against the collection `/api/coffee` and resource actions are those corresponding to a single resource `/api/coffee/:c_id`
+ 
+Through the above definition, the following endpoints and actions were created:
+```
+GET /api/coffee
+GET /api/coffee/:c_id
+
+POST /api/coffee
+POST /api/coffee { "action": "edit" }
+POST /api/coffee { "action": "delete" }
+POST /api/coffee/:c_id { "action": "edit" } 
+POST /api/coffee/:c_id { "action": "delete" } 
+
+DELETE /api/coffee/:c_id
+```
+### **Where are the methods?**
+Now that the routes exist, let's take care of the methods. Once the new collection has been added to the `api.yml`, the corresponding controller needs to be [created here](https://github.com/ManageIQ/manageiq-api/tree/4811ec9fcea2424a585560a761e05d56c53535f8/app/controllers/api). The controller will look something like:
+```
+module Api
+  class CoffeeController < BaseController 
+  end 
+end 
+```
+Because the `CoffeeController` inherits from the `BaseController`, a lot of what has been defined in the `api.yml` will already work :sparkles:. The [Base Controller](https://github.com/ManageIQ/manageiq-api/blob/4811ec9fcea2424a585560a761e05d56c53535f8/app/controllers/api/base_controller.rb) holds the action methods (`#show`, `#index`, `#create`, `#update`, `#destroy`) that will be called if they are not overridden in the `CoffeeController`.
```

Comment:
We don't have a `#create` action yet, but funnily enough I was just reviving an old branch to add one 😆